src/parsers/perf/perfparser.cpp

```diff
@@ -1351,46 +1395,71 @@ void PerfParser::startParseFile(const QString& path, const QString& sysroot, con
         return;
     }
 
-    QStringList parserArgs = {QStringLiteral("--input"), decompressIfNeeded(path), QStringLiteral("--max-frames"),
-                              QStringLiteral("1024")};
-    if (!sysroot.isEmpty()) {
-        parserArgs += {QStringLiteral("--sysroot"), sysroot};
-    }
-    if (!kallsyms.isEmpty()) {
-        parserArgs += {QStringLiteral("--kallsyms"), kallsyms};
-    }
-    if (!debugPaths.isEmpty()) {
-        parserArgs += {QStringLiteral("--debug"), debugPaths};
-    }
-    if (!extraLibPaths.isEmpty()) {
-        parserArgs += {QStringLiteral("--extra"), extraLibPaths};
-    }
-    if (!appPath.isEmpty()) {
-        parserArgs += {QStringLiteral("--app"), appPath};
-    }
-    if (!arch.isEmpty()) {
-        parserArgs += {QStringLiteral("--arch"), arch};
-    }
+    auto parserArgs = [this](const QString& filename){
+        const auto settings = Settings::instance();
+
+        QStringList parserArgs = {QStringLiteral("--input"), decompressIfNeeded(filename), QStringLiteral("--max-frames"),
+                                  QStringLiteral("1024")};
+        const auto sysroot = settings->sysroot();
+        if (!sysroot.isEmpty()) {
+            parserArgs += {QStringLiteral("--sysroot"), sysroot};
+        }
+        const auto kallsyms = settings->kallsyms();
+        if (!kallsyms.isEmpty()) {
+            parserArgs += {QStringLiteral("--kallsyms"), kallsyms};
+        }
+        const auto debugPaths = settings->debugPaths();
+        if (!debugPaths.isEmpty()) {
+            parserArgs += {QStringLiteral("--debug"), debugPaths};
+        }
+        const auto extraLibPaths = settings->extraLibPaths();
+        if (!extraLibPaths.isEmpty()) {
+            parserArgs += {QStringLiteral("--extra"), extraLibPaths};
+        }
+        const auto appPath = settings->appPath();
+        if (!appPath.isEmpty()) {
+            parserArgs += {QStringLiteral("--app"), appPath};
+        }
+        const auto arch = settings->arch();
+        if (!arch.isEmpty()) {
+            parserArgs += {QStringLiteral("--arch"), arch};
+        }
+        return parserArgs;
+    };
 
     // reset the data to ensure filtering will pick up the new data
-    m_parserArgs = parserArgs;
+    m_parserArgs = parserArgs(path);
     m_bottomUpResults = {};
     m_callerCalleeResults = {};
     m_events = {};
 
+    auto env = Util::appImageEnvironment();
```

Comment:
this part here also seems to be a general cleanup commit that we should do independent of this patch. it's actually fixing a potential race condition, as the settings instance isn't thread safe - ideally you'd also add this assert and fix the fallout:

```
Settings *Settings::instance()
{
    Q_ASSERT(QThread::currentThread() == QCoreApplication::instance()->thread());
    return ...;
}
```